From af629960be92b1548d0b3c15b6e082930403fbae Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Fri, 10 Jul 2020 16:57:43 +0100
Subject: [PATCH] SQUASH: dts: Further simplify firmware clocks

All Pi platforms will use the firmware clocks driver, so declare it in
the most common place - bcm2835-rpi.dtsi.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2709-rpi.dtsi | 7 -------
 arch/arm/boot/dts/bcm2711-rpi.dtsi | 7 -------
 arch/arm/boot/dts/bcm2835-rpi.dtsi | 5 +++++
 arch/arm/boot/dts/bcm2836-rpi.dtsi | 7 -------
 4 files changed, 5 insertions(+), 21 deletions(-)

--- a/arch/arm/boot/dts/bcm2709-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2709-rpi.dtsi
@@ -3,10 +3,3 @@
 &vchiq {
 	compatible = "brcm,bcm2836-vchiq", "brcm,bcm2835-vchiq";
 };
-
-&firmware {
-	firmware_clocks: clocks {
-		compatible = "raspberrypi,firmware-clocks";
-		#clock-cells = <1>;
-	};
-};
--- a/arch/arm/boot/dts/bcm2711-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2711-rpi.dtsi
@@ -309,10 +309,3 @@
 &hvs {
 	clocks = <&firmware_clocks 4>;
 };
-
-&firmware {
-	firmware_clocks: clocks {
-		compatible = "raspberrypi,firmware-clocks";
-		#clock-cells = <1>;
-	};
-};
--- a/arch/arm/boot/dts/bcm2835-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2835-rpi.dtsi
@@ -16,6 +16,11 @@
 			compatible = "raspberrypi,bcm2835-firmware", "simple-bus";
 			mboxes = <&mailbox>;
 			dma-ranges;
+
+			firmware_clocks: clocks {
+				compatible = "raspberrypi,firmware-clocks";
+				#clock-cells = <1>;
+			};
 		};
 
 		power: power {
--- a/arch/arm/boot/dts/bcm2836-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2836-rpi.dtsi
@@ -4,10 +4,3 @@
 &vchiq {
 	compatible = "brcm,bcm2836-vchiq", "brcm,bcm2835-vchiq";
 };
-
-&firmware {
-	firmware_clocks: clocks {
-		compatible = "raspberrypi,firmware-clocks";
-		#clock-cells = <1>;
-	};
-};
